torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -6587,6 +6587,21 @@ std::tuple<Tensor, Tensor> index_reduce_backward(
   return std::make_tuple(grad_self, grad_src);
 }
 
+Tensor take_backward(
+    const Tensor& grad,
+    const Tensor& self,
+    const Tensor& indices) {
+  Tensor grad_self = self.new_zeros(self.sizes());
```

Comment:
this is just `at::zeros_like(self)`, right? there's no real advantage to doing `self.new_zeros(self.sizes())` here

Reply:
We have a fast-path right below this code which considers depends on `self.new_zeros`.
If `self` is Subclass, then we can inplace into `grad_self` irrespective of `grad` or `indices` being Subclass.

Reply:
I believed that `at::zeros_like(self` would return a regular tensor.